test/json.js

```diff
@@ -14,79 +14,140 @@
  */
 
 /* jslint node: true */
-/* global describe: false, before: false, after: false, it: false */
 'use strict';
 
-
-var NATS = require('../'),
+const NATS = require('../'),
     nsc = require('./support/nats_server_control'),
     should = require('should');
 
-describe('JSON payloads', function() {
-
-    var PORT = 1423;
-    var server;
+describe('JSON payloads', () => {
+    const PORT = 1423;
+    let server;
 
     // Start up our own nats-server
-    before(function(done) {
+    before((done) => {
         server = nsc.start_server(PORT, done);
     });
 
     // Shutdown our server
-    after(function(done) {
+    after((done) => {
         nsc.stop_server(server, done);
     });
 
+    function testPubSub({ input, expected }) {
```

Comment:
While we are not testing on node 4.5, our source is compatible and works correctly on node 4.5+. The object destructuring is wicked cool, but not supported in nodejs4.